0.preprocess-sites/4.image-and-segmentation-qc.py

```diff
@@ -248,90 +268,105 @@
             if "Mask" in col:
                 corr_qc_col = col
 image_df_subset = image_df.dropna(subset=[corr_qc_col])
-correlation_gg = (
-    gg.ggplot(image_df_subset, gg.aes(x="x_loc", y="y_loc"))
-    + gg.geom_point(gg.aes(fill=corr_qc_col), shape="s", size=6)
-    + gg.geom_text(gg.aes(label="site_location"), color="lightgrey", size=6)
-    + gg.facet_wrap("~Metadata_Well")
-    + gg.coord_fixed()
-    + gg.theme_bw()
-    + gg.ggtitle(
-        f"Correlation between BC & CP DAPI images\n(Alignment quality)\n{plate}"
-    )
-    + gg.theme(
-        axis_text=gg.element_blank(),
-        axis_title=gg.element_blank(),
-        strip_background=gg.element_rect(colour="black", fill="#fdfff4"),
-    )
-    + gg.scale_fill_cmap(name="Correlation")
-)
-
-output_file = pathlib.Path(
-    output_figuresdir, f"plate_layout_BC_to_CP_DAPI_correlation_per_well.png",
-)
-if check_if_write(output_file, force, throw_warning=True):
-    correlation_gg.save(output_file, dpi=300, verbose=False)
-
-# Plot final compartment thresholds per well
-threshold_col_prefix = "Threshold_FinalThreshold_"
-for threshhold_compartment in ["Cells", "Nuclei"]:
-    threshold_full_col = f"{threshold_col_prefix}{threshhold_compartment}"
-
-    if threshold_full_col not in image_df.columns:
-        continue
-
-    image_df_subset = image_df.dropna(subset=[threshold_full_col])
-
-    compartment_finalthresh_gg = (
-        gg.ggplot(image_df_subset, gg.aes(x="x_loc", y="y_loc"))
-        + gg.geom_point(gg.aes(fill=f"{threshold_full_col}"), shape="s", size=6)
+for plate in platelist:
+    correlation_gg = (
+        gg.ggplot(
+            image_df_subset.loc[image_df_subset["Metadata_Plate"].str.contains(plate)],
+            gg.aes(x="x_loc", y="y_loc"),
+        )
+        + gg.geom_point(gg.aes(fill=corr_qc_col), shape="s", size=6)
         + gg.geom_text(gg.aes(label="site_location"), color="lightgrey", size=6)
-        + gg.facet_wrap(f"~{image_cols['well']}")
+        + gg.facet_wrap("~Metadata_Well")
```

Comment:
i think this needs to change back to `image_cols["well"]` to avoid hardcoding, right?

Reply:
Oops. Fixed that and other instances of metadata column names being hardcoded in latest commit 6513f7a

Reply:
nice! Thanks for providing the commit hash - looks like there were lots of these fixes required